x-pack/plugins/security_solution/public/management/components/endpoint_responder/hooks.tsx

```diff
@@ -0,0 +1,91 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { useEffect } from 'react';
+import { useIsMounted } from '../../hooks/use_is_mounted';
+import { useGetActionDetails } from '../../hooks/endpoint/use_get_action_details';
+import { ACTION_DETAILS_REFRESH_INTERVAL } from './constants';
+import type { ActionRequest, ActionRequestComponentProps } from './types';
+import type { useSendIsolateEndpointRequest } from '../../hooks/endpoint/use_send_isolate_endpoint_request';
+import type { useSendReleaseEndpointRequest } from '../../hooks/endpoint/use_send_release_endpoint_request';
+
+export const useUpdateActionState = ({
+  actionRequestApi,
+  actionRequest,
+  command,
+  endpointId,
+  setStatus,
+  setStore,
+  isPending,
+}: Pick<ActionRequestComponentProps, 'command' | 'setStatus' | 'setStore'> & {
+  actionRequestApi: ReturnType<
```

Comment:
Looks like the only difference in the return types would be the request params from the API... You could make this even more generic here by making this hook a generic and removing the `typeof <hookname>` below- like:

```typescript
export interface UpdateActionStateProps {
  //...

  actionRequestApi<T = unknown>: seMutationResult<
    ResponseActionApiResponse,
    IHttpFetchError,
    T
  >
}

export const useUpdateActionState<T>(args: UpdateActionStateProps<T>) {...
```

Reply:
I just realized that this PR is addressing only Isolate/Release, so you can defer this change and the one about passing in the `body` for the action request api call

Reply:
Yeah, I wanted to refactor the other actions as well, but for now we only need to deal with the isolation status for the endpoint. Also, the other actions have API response type (that returns a `data.id` instead of `action`) and error handling that needs to be uniform for isolate and release. Best to do that in a new PR.

Reply:
FYI - I thought that the release/isolate API also has the `data.id` now, so you should be able to just use that across all actions

Reply:
3389576eebb1083b6d591c355fd6a8716eef9dfb